src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -1,44 +1,213 @@
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventQueryEndpointImpl  implements EventQueryEndpoint {
+
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    private static final String JSON_PARSE_ERROR_MESSAGE = "cannot parse request body to event query";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.post(EventQueryEndpoint.getCreateEventQueryBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::createEventQuery));
+
+        sparkService.delete(EventQueryEndpoint.getDeleteEventQueryBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::deleteEventQuery));
+
+        sparkService.put(EventQueryEndpoint.getEditEventQueryBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::editEventQuery));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventQuery(Request request, Response response) {
+		JsonObject jsonEventQuery = jsonParser.parse(request.body()).getAsJsonObject();
+
+		if (jsonEventQuery == null) {
+			halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+		}
+
+		EventQuery eventQuery = getEventQueryFromJson(jsonEventQuery, true);
+		if (eventQuery == null) {
+			halt(HttpStatusCodes.BAD_REQUEST, "cannot parse event query");
+		}
+
+		checkEventTypeExists(eventQuery);
+
+		EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventQuery(eventQuery.getTypeId(), eventQuery);
+
+		if (!feedback.isSuccessful()) {
+			halt(HttpStatusCodes.ERROR, String.format("cannot register event type: %1$s", feedback.getResponseText()));
+		}
+
+		PersistenceAdapterImpl.getInstance().createArtifact(eventQuery, EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()));
+
+		return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given query was not found");
+        }
+
+		EventType queryOwner = PersistenceAdapterImpl.getInstance().getEventType(eventQuery.getTypeId());
+        if (queryOwner == null) {
+        	halt(HttpStatusCodes.ERROR, "cannot find query owner");
+		}
+
+		if (!queryOwner.isDeletable()

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'queryOwner' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)